samples/bluetooth/hci_spi/src/main.c

```diff
@@ -195,6 +195,7 @@ static void bt_tx_thread(void *p1, void *p2, void *p3)
 
 		switch (rxmsg[PACKET_TYPE]) {
 		case HCI_CMD:
+			memcpy(&cmd_hdr, &rxmsg[1], sizeof(cmd_hdr));
```

Comment:
Coping seems uneccesary, I think changing cmd_hdr and acl_hdr to pointers into rxmsg would be better.

```
		case HCI_CMD:
		{
			struct bt_hci_cmd_hdr *cmd_hdr;

			buf = bt_buf_get_tx(BT_BUF_CMD, K_NO_WAIT, &rxmsg[1],
					    sizeof(*cmd_hdr));
			if (buf) {
				net_buf_add_mem(buf, &rxmsg[4],
						cmd_hdr->param_len);
...
```

Reply:
I agree I'll change that !
I initially put the original implementation.

Reply:
That's a clever way of avoiding copies. Else we could have just get rid of the header structs and reference the buffer directly and use the header size defines instead of the sizeof(cmd_hdr).

But I like this approach more as it is straight forward compatible with v2.3.0.